src/Character.js

```diff
@@ -60,6 +61,26 @@ class Character extends EventEmitter
    */
   getMaxAttribute(attr) {
     const attribute = this.attributes.get(attr);
+
+    // health and attackpower are calculated based off other stats
+    // NOTE: This is an _example_ implementation based off WoW formulas
+    switch (attribute.name) {
+      case 'health':
+        attribute.setBase(
+          this.getMaxAttribute('stamina') * healthPerStamina(this.level)
+        );
+        break;
+      case 'attackpower':
+        attribute.setBase(this.getMaxAttribute('strength'));
```

Comment:
That's not exactly what's going on. Nested function calls may make it harder to read, but here is what's going on:

```
const strength = this.getMaxAttribute('strength'); // This will return the value of their strength attribute as modified by effects. E.g., 18.
attribute.setBase(strength); // This will set the current attribute (attackpower?) to 18.
```